crates/uv-interpreter/src/python_query.rs

```diff
@@ -476,34 +474,47 @@ mod windows {
             .collect())
     }
 
-    /// On Windows we might encounter the windows store proxy shim (Enabled in Settings/Apps/Advanced app settings/App execution aliases).
-    /// This requires quite a bit of custom logic to figure out what this thing does.
+    /// On Windows we might encounter the Windows Store proxy shim (enabled in:
+    /// Settings/Apps/Advanced app settings/App execution aliases). When Python is _not_ installed
+    /// via the Windows Store, but the proxy shim is enabled, then executing `python.exe` or
+    /// `python3.exe` will redirect to the Windows Store installer.
+    ///
+    /// We need to detect that these `python.exe` and `python3.exe` files are _not_ Python
+    /// executables.
+    ///
+    /// This method is taken from Rye:
     ///
-    /// This is a pretty dumb way.  We know how to parse this reparse point, but Microsoft
-    /// does not want us to do this as the format is unstable.  So this is a best effort way.
-    /// we just hope that the reparse point has the python redirector in it, when it's not
-    /// pointing to a valid Python.
+    /// > This is a pretty dumb way.  We know how to parse this reparse point, but Microsoft
+    /// > does not want us to do this as the format is unstable.  So this is a best effort way.
+    /// > we just hope that the reparse point has the python redirector in it, when it's not
+    /// > pointing to a valid Python.
     ///
-    /// Matches against paths like:
-    ///     `C:\Users\crmar\AppData\Local\Microsoft\WindowsApps\python.exe`
+    /// See: <https://github.com/astral-sh/rye/blob/b0e9eccf05fe4ff0ae7b0250a248c54f2d780b4d/rye/src/cli/shim.rs#L108>
+    #[cfg(windows)]
     pub(super) fn is_windows_store_shim(path: &Path) -> bool {
-        // Rye uses a more sophisticated test to identify the windows store shim.
-        // Unfortunately, it only works with the `python.exe` shim but not `python3.exe`.
-        // What we do here is a very naive implementation but probably sufficient for all we need.
-        // There's the risk of false positives but I consider it rare, considering how specific
-        // the path is.
+        use std::os::windows::fs::MetadataExt;
+        use std::os::windows::prelude::OsStrExt;
+        use winapi::um::fileapi::{CreateFileW, OPEN_EXISTING};
+        use winapi::um::handleapi::{CloseHandle, INVALID_HANDLE_VALUE};
+        use winapi::um::ioapiset::DeviceIoControl;
+        use winapi::um::winbase::{FILE_FLAG_BACKUP_SEMANTICS, FILE_FLAG_OPEN_REPARSE_POINT};
+        use winapi::um::winioctl::FSCTL_GET_REPARSE_POINT;
+        use winapi::um::winnt::{FILE_ATTRIBUTE_REPARSE_POINT, MAXIMUM_REPARSE_DATA_BUFFER_SIZE};
+
+        // The path must be absolute.
         if !path.is_absolute() {
             return false;
         }
 
+        // The path must point to something like:
+        //   `C:\Users\crmar\AppData\Local\Microsoft\WindowsApps\python3.exe`
```

Comment:
Can we reuse the uv-fs function instead of reimplementing the check?

Reply:
They're actually different checks (sigh). The one in `uv-fs` needs to accept `python3.12.exe` and friends. The one here only accepts `python3.exe` and `python.exe`, since those are the only ones that get added as app redirects.